### PR TITLE
Adds 'status' field to Delta object

### PR DESCRIPTION
**Pull request by arnav-mandal1234**
This adds a new status field in Delta class
Fixes : https://github.com/nexB/deltacode/issues/110
Signed-off-by: Arnav Mandal <arnav.mandal1234@gmail.com>

### CODE REVIEW

**Comment by arnav-mandal1234** [comment]
@majurg @pombredanne Kindly review this.
This is my first PR here. :)

File: src/deltacode/__init__.py [hunk 1]
```diff
@@ -106,6 +106,7 @@ def determine_delta(self):
                 except KeyError:
                     delta = Delta(100, new_file, None)
                     delta.factors.append('added')
```
Comment (steven-esser) [inline src/deltacode/__init__.py]: Now that we have a `status` field for the delta object, we do not need to append that string to our `factors` field.

File: src/deltacode/__init__.py [hunk 2]
```diff
@@ -117,11 +118,13 @@ def determine_delta(self):
                     if new_file.sha1 == f.sha1:
                         delta = Delta(0, new_file, f)
                         delta.factors.append('unmodified')
```
Comment (steven-esser) [inline src/deltacode/__init__.py]: Now that we have a `status` field for the delta object, we do not need to append that string to our `factors` field.

File: src/deltacode/__init__.py [hunk 3]
```diff
@@ -138,6 +141,7 @@ def determine_delta(self):
                 except KeyError:
                     delta = Delta(0, None, old_file)
                     delta.factors.append('removed')
```
Comment (steven-esser) [inline src/deltacode/__init__.py]: Now that we have a `status` field for the delta object, we do not need to append that string to our `factors` field.

File: src/deltacode/__init__.py [hunk 4]
```diff
@@ -181,6 +185,7 @@ def update_deltas(self, added, removed):
         """
         delta = Delta(0, added.new_file, removed.old_file)
         delta.factors.append('moved')
```
Comment (steven-esser) [inline src/deltacode/__init__.py]: Now that we have a `status` field for the delta object, we do not need to append that string to our `factors` field.

**Review by steven-esser — CHANGES_REQUESTED** [review]
This is the right direction. See specific comments for needed changes.

**Comment by steven-esser** [comment]
@arnav-mandal1234 Yeah it looks like MacOS builds are hanging for `scancodle-toolkit` as well. 

I will look into this.

File: tests/data/deltacode/scan_sorted01_new.json [hunk 5]
```diff
@@ -86,7 +59,7 @@
       "is_script": false,
       "licenses": [
         {
-          "key": "apache-2.0",
+          "key": "apache-3.0",
```
Comment (steven-esser) [inline tests/data/deltacode/scan_sorted01_new.json:62]: why is this changed to apache-3.0?

Comment (arnav-mandal1234) [reply tests/data/deltacode/scan_sorted01_new.json:62]: @majurg I just wanted my factor list to be filled with "license change". Previously factors list was filled with status, so to make the test method work, I have to add other factors excluding status.

Comment (steven-esser) [reply tests/data/deltacode/scan_sorted01_new.json:62]: @arnav-mandal1234 OK that makes sense.

**Review by steven-esser — APPROVED** [review]
